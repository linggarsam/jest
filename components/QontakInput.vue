<template>
  <b-input-group>
    <b-form-input
      autocomplete="off"
      :value="value"
      @input="onInput"
    />
  </b-input-group>
</template>

<script>
export default {
  props: {
    value: {
      type: String,
      default: () => ''
    }
  },
  methods: {
    onInput (val) {
      this.$emit('input', val)
    }
  }
}
</script>
