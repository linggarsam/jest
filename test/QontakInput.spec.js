import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import QontakInput from '@/components/QontakInput.vue'
import * as BootstrapVue from 'bootstrap-vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

test('Test mounted', () => {
  const wrapper = mount(QontakInput, {
    localVue
  })
  expect(wrapper.vm).toBeTruthy()
});

test('Test snapshot', () => {
  const wrapper = mount(QontakInput, {
    localVue
  })
  expect(wrapper.html()).toMatchSnapshot()
});

test('Test emitter onInput', () => {
  const wrapper = shallowMount(QontakInput, {
    localVue
  })
  wrapper.vm.onInput('Oke')
  expect(wrapper.emitted().input).toEqual([['Oke']])
})


