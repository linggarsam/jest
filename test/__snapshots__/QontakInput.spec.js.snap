// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
"<div role=\\"group\\" class=\\"input-group\\">
  <!----><input type=\\"text\\" autocomplete=\\"off\\" class=\\"form-control\\">
  <!---->
</div>"
`;
